### PR TITLE
fix: Disabled claim till generate new

### LINKED ISSUE #8103: Mint nft button disabled and enabled states
It seems like its not working 100% 
- I opened the drop page, and the button was disabled even though I hadn't minted yet. This could be a cookie issue, but still worth fixing because I am not the only one that minted the previous drop 
- I reconnected a different wallet, minted nft and the button is still enabled - should be disabled state

image

image

### LINKED ISSUE #8315: Disabled claim till generate new
when you land the ui, the button is disabled till you click Generate new 

image
image

tested on canary

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Bugfix

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #8315
  - [x] Closes #8103
  - [ ] Requires deployment <snek/rubick/worker>

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [ ] My fix has changed UI

  ## Copilot Summary
  
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 4039f5c</samp>

Refactored `GenerativePreview.vue` to use a named function for generating default NFTs on mount and account change.

  
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 4039f5c</samp>

> _`generateDefaultNft`_
> _refactors `generateNft` -_
> _clear code in springtime_

### CODE REVIEW

**Comment by exezbcz** [comment]
I know that, on one side, it could help remove the duplicate drops that happen because users have the same nft for minting. 
- when we active the button after generate new

It will be solved by iteration of the drop page and one followup


Works for me, drop for testing would be nice :D

File: components/collection/drop/GenerativePreview.vue [hunk 1]
```diff
@@ -77,9 +77,12 @@ const generativeImageUrl = ref(
 
 const isLoading = ref(false)
 
-watch([accountId], () => {
+const generateDefaultNft = () => {
   generateNft(true)
-})
```
Comment (preschian) [inline components/collection/drop/GenerativePreview.vue:82]: use `{ immediate: true }` here? or `watchEffect`

**Review by preschian — APPROVED** [review]

**Comment by yangwao** [comment]
> Works for me, drop for testing would be nice :D

@vikiival may set some testing drop on AHK for this. As we would need to cook few things till Thursday anyway.

**Comment by vikiival** [comment]
As @yangwao pinged me, I made a testing collection on AHK (id 290)

Screenshot 2023-12-05 at 11 41 08

**Pull request merged**